EventFilter/CTPPSRawToDigi/src/RPixErrorChecker.cc

```diff
@@ -73,57 +93,111 @@ bool RPixErrorChecker::checkTrailer(bool& errorsInEvent, int fedId, unsigned int
   if ( fedTrailer.fragmentLength()!= nWords) {
     LogError("FedTrailerLenght")<< "fedTrailer.fragmentLength()!= nWords !! Fed: " << fedId << ", errorType = 34";
     errorsInEvent = true;
- 
+    if(includeErrors) {
+      int errorType = 34;
+      CTPPSPixelDataError error(*trailer, errorType, fedId);
+      errors[dummyDetId].push_back(error);
+    }
   }
   return fedTrailer.moreTrailers();
 }
 
-bool RPixErrorChecker::checkROC(bool& errorsInEvent, int fedId,  Word32& errorWord) const
+bool RPixErrorChecker::checkROC(bool& errorsInEvent, int fedId, uint32_t iD, Word32& errorWord,  Errors& errors) const
 {
   int errorType = (errorWord >> ROC_shift) & ERROR_mask;
   if likely(errorType<25) return true;
 
   switch (errorType) {
   case(25) : {
-    LogDebug("")<<"  invalid ROC=25 found (errorType=25)";
+    LogError("")<<"  invalid ROC=25 found (errorType=25)";
```

Comment:
is this change "safe" in a sense that do we expect these to be issued only rarely?

Reply:
Yes, the idea is that it should be issued rarely, better if never at all, but sending a clear error message, not only if in debug mode.

Reply:
Given that this code also creates a special persisted error collection, it seems like the additional error message is a redundant information.
I still think that this should be disabled in production logs.

Reply:
Ok. Done.